salt/modules/file.py

```diff
@@ -2093,14 +2093,10 @@ def replace(path,
                         # Content was found, so set found.
                         found = True
 
-                # Keep track of show_changes here, in case the file isn't
-                # modified
-                if show_changes or append_if_not_found or \
-                   prepend_if_not_found:
-                    orig_file = r_data.read(filesize).splitlines(True) \
-                        if isinstance(r_data, mmap.mmap) \
-                        else r_data.splitlines(True)
-                    new_file = result.splitlines(True)
+                orig_file = r_data.read(filesize).splitlines(True) \
```

Comment:
Ahh, no. Of course it wasn't _broken_ on purpose. But yes, the purpose was to avoid keeping two copies in memory for every use case. And yes, there is such a warning; just [look at the docstring](https://github.com/saltstack/salt/blob/develop/salt/modules/file.py#L1960-L1964).